### ISSUE TITLE
model output changes every time when use onnx model

### ISSUE THREAD

**Issue by HaoLiuHust** [NVIDIA/TensorRT#493]
## Description

I have converted a pytorch model to onnx format, and I have checked pytorch output is same with onnx, so the converted process should be ok. Then I use this onnx model with tensorrt in sampleOnnxMnist(with some modification since the input changed), but I found that with same input, the output changes every time I rerun the model. I have run this mode in  sampleOnnxMnist, the 

## Environment

**TensorRT Version**: 7.0.0.11
**GPU Type**: 1080Ti
**Nvidia Driver Version**: 390.77
**CUDA Version**: 9.0
**CUDNN Version**: 7.6.5
**Operating System + Version**: ubuntu 16.04
**Python Version (if applicable)**: 
**TensorFlow Version (if applicable)**: 
**PyTorch Version (if applicable)**: 
**Baremetal or Container (if container which image + tag)**: 

## Relevant Files
[sampleOnnxMNIST.zip](https://github.com/NVIDIA/TensorRT/files/4473546/sampleOnnxMNIST.zip)

## Steps To Reproduce
1. unzip the zip file  
2. put onnx model to tensorrt install folder/data/mnist  
3. replace sampleOnnxMNIST

**Comment by HaoLiuHust** [comment]
@rmccorm4 could you help?

**Comment by rmccorm4** [comment]
Hi @HaoLiuHust ,

There's probably a bug in your code. I'd double check all of the buffers and everything are setup correctly.

I wrote a simple python script to test your model instead.

Here's the environment:
```
# TensorRT 7.0 container
nvidia-docker run -it -v ${PWD}:/mnt --workdir=/mnt nvcr.io/nvidia/tensorrt:20.03-py3
# PyTorch 1.4
pip install torch==1.4.0
```

Here's the script:
```python
import torch
import numpy as np
import tensorrt as trt
import pycuda.autoinit

TRT_LOGGER = trt.Logger(trt.Logger.INFO)

def build_engine(onnx_model):
    EXPLICIT_BATCH = 1 << int(trt.NetworkDefinitionCreationFlag.EXPLICIT_BATCH)
    with trt.Builder(TRT_LOGGER) as builder, \
         builder.create_builder_config() as config, \
         builder.create_network(EXPLICIT_BATCH) as network, \
         trt.OnnxParser(network, TRT_LOGGER) as parser:

        with open(onnx_model, "rb") as f:
            if not parser.parse(f.read()):
                print('ERROR: Failed to parse the ONNX file: {}'.format(args.onnx))
                for error in range(parser.num_errors):
                    print(parser.get_error(error))
                sys.exit(1)

        # Builder Config flags go here
        return builder.build_engine(network, config)

def infer(context, bindings):
    context.execute_v2(bindings)

def compare(results):
    for i in range(1, len(results)):
        np.testing.assert_almost_equal(results[i], results[i-1])

if __name__ == "__main__":
    print("Build")
    engine = build_engine("reid_osnet_static.onnx")

    print("Setup")
    inputs = [torch.ones((1, 3, 256, 128), device="cuda:0")]
    outputs = [torch.zeros((1, 512), device="cuda:0")]
    bindings = [_input.data_ptr() for _input in inputs] + [_output.data_ptr() for _output in outputs]

    print("Create context")
    context = engine.create_execution_context()

    print("Infer")
    results = []
    for i in range(10):
        infer(context, bindings)
        results.append(outputs[0].cpu().numpy().copy())

    print("Compare")
    compare(results)

    print("All values match within tolerance.")
```

**Comment by HaoLiuHust** [comment]
@rmccorm4 yes, I have test python too, it's ok

**Comment by HaoLiuHust** [comment]
I was using sampleOnnxMnist, and the only change is replace the onnx model

**Comment by rmccorm4** [comment]
Try writing a simpler C++ example instead of using the MNIST sample. The Python API is just a wrapper around C++ API, so results should be the same.

**Comment by HaoLiuHust** [comment]
I am writing my simpler C++ example，but encounter a strange problem:
```
std::shared_ptr<nvinfer1::ICudaEngine> BuildEngine(const string& onnx_model)
{
	Logger2 gLogger;

	auto builder = SampleUniquePtr<nvinfer1::IBuilder>(nvinfer1::createInferBuilder(gLogger));
	if (!builder)
	{
		return nullptr;
	}

	const auto explicitBatch = 1U << static_cast<uint32_t>(nvinfer1::NetworkDefinitionCreationFlag::kEXPLICIT_BATCH);
	auto network = SampleUniquePtr<nvinfer1::INetworkDefinition>(builder->createNetworkV2(explicitBatch));
	if (!network)
	{
		return nullptr;
	}

	auto config = SampleUniquePtr<nvinfer1::IBuilderConfig>(builder->createBuilderConfig());
	if (!config)
	{
		return nullptr;
	}

	auto parser = SampleUniquePtr<nvonnxparser::IParser>(nvonnxparser::createParser(*network, gLogger));
	if (!parser)
	{
		return nullptr;
	}

	auto parsed = parser->parseFromFile(
			onnx_model.c_str(), 1);
	if (!parsed)
	{
		return nullptr;
	}

	builder->setMaxBatchSize(1);
	config->setMaxWorkspaceSize(16* 1<<20);

	auto mEngine = std::shared_ptr<nvinfer1::ICudaEngine>(
			builder->buildEngineWithConfig(*network, *config), InferDeleter());
	if (!mEngine)
	{
		return nullptr;
	}

	// we don't need the network any more, and we can destroy the parser

	return mEngine;
}

int main()
{
	cudaSetDevice(0);
	auto engine = BuildEngine("/home/liuhao/software/Tensorrt/TensorRT-7.0.0.11/data/mnist/reid_osnet_static.onnx");

	if(!engine)
	{
		printf("error\n");
		exit(-1);
	}

	auto context = engine->createExecutionContext();

//	//create bindings
//	int nb_bindings = engine->getNbBindings();
//	void** buffers = (void **)malloc(sizeof(float *) * nb_bindings);
//	for (int b = 0; b < nb_bindings; b++) {
//		Dims dims = engine->getBindingDimensions(b);
//		int nbDims = dims.nbDims;
//		size_t size = 1;
//		for (int i = 1; i < nbDims; i++)
//			size *= dims.d[i];
//
//		cudaMalloc(&buffers[b], size * sizeof(float));
//	}


	context->destroy();
}
```
it will be crashed in 
```
auto context = engine->createExecutionContext();
```
but if I put this line into BuildEngine, it will be ok. How could this be happen?

**Comment by HaoLiuHust** [comment]
seems the logger should be move out the BuildEngine function...，if so, it will be ok

**Comment by HaoLiuHust** [comment]
test it for one day...I think the reason is the data is not normalized, if I normalize the data before input to model, it will be ok
